libqtile/widget/check_updates.py

```diff
@@ -78,3 +79,8 @@ def poll(self):
         if not self.cmd:
             return "N/A"
         return self._check_updates()
+
+    def button_press(self, x, y, button):
+        base.ThreadedPollText.button_press(self, x, y, button)
+        if button == 1 and self.execute is not None:
+            Popen([self.execute], shell=True)
```

Comment:
for now I use 'urxvt -e bash -c "yaourt -Syua"' and this works. May be I
should check is it list, overwise split it to list?


On 14 July 2015 at 17:27, Tycho Andersen <notifications@github.com> wrote:

> In libqtile/widget/check_updates.py
> <https://github.com/qtile/qtile/pull/699#discussion_r34573256>:
>
> > @@ -78,3 +79,8 @@ def poll(self):
> >          if not self.cmd:
> >              return "N/A"
> >          return self._check_updates()
> > +
> > +    def button_press(self, x, y, button):
> > +        base.ThreadedPollText.button_press(self, x, y, button)
> > +        if button == 1 and self.execute is not None:
> > +            Popen([self.execute], shell=True)
>
> You might make this just self.execute here, so that people can pass in a
> list if their distro update command (i.e. apt-get deselect-upgrade) is
> more than one word.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/qtile/qtile/pull/699/files#r34573256>.
>



-- 
Udovīčė Arns